Build failure with `HUnit < 1.5`

```
$ cabal build -w ghc-8.2.2 -c 'HUnit < 1.5'
In order, the following will be built (use -v for more details):
 - hspec-expectations-0.8.4 (lib) (first run)
...
src/Test/Hspec/Expectations/Contrib.hs:16:43: error:
    Module ‘Test.HUnit.Lang’ does not export ‘FailureReason(..)’
   |
16 | import Test.HUnit.Lang (HUnitFailure(..), FailureReason(..))
   |                                           ^^^^^^^^^^^^^^^^^
```
In the wild: https://github.com/haskell-hvr/http-io-streams/actions/runs/5597341206/jobs/10235560859#step:17:636

--- a/hspec-expectations.cabal
+++ b/hspec-expectations.cabal
@@ -27,7 +27,7 @@ library
       src
   ghc-options: -Wall
   build-depends:
-      HUnit
+      HUnit >=1.5.0.0
     , base ==4.*
     , call-stack
   exposed-modules:
@@ -47,7 +47,7 @@ test-suite spec
   ghc-options: -Wall
   build-depends:
       HUnit >=1.5.0.0
-    , base ==4.*
+    , base >=4.8
     , call-stack
     , nanospec
   other-modules:

--- a/package.yaml
+++ b/package.yaml
@@ -15,11 +15,10 @@ ghc-options: -Wall
 dependencies:
   - base == 4.*
   - call-stack
+  - HUnit >= 1.5.0.0
 
 library:
   source-dirs: src
-  dependencies:
-    - HUnit
   exposed-modules:
     - Test.Hspec.Expectations
     - Test.Hspec.Expectations.Contrib
@@ -31,5 +30,5 @@ tests:
       - test
       - src
     dependencies:
+      - base >= 4.8
       - nanospec
-      - HUnit >= 1.5.0.0

